packages/clima_ui/lib/screens/weather_screen.dart

```diff
@@ -32,32 +35,56 @@ class WeatherScreen extends HookWidget {
 
     final fullWeatherStateNotifier =
         useProvider(w.fullWeatherStateNotifierProvider.notifier);
+
     final controller = useFloatingSearchBarController();
 
-    final isLoading = useState(fullWeatherState is c.Loading);
+    final isCityChanging = useState(false);
 
     final cityStateNotifier = useProvider(c.cityStateNotifierProvider.notifier);
 
-    Future<void> load() async {
-      await fullWeatherStateNotifier.loadFullWeather();
-    }
+    final refreshIndicatorKey = useGlobalKey<RefreshIndicatorState>();
 
     useEffect(
-      () => cityStateNotifier.addListener((state) {
-        if (state is c.Error) {
-          showFailureSnackBar(context, failure: state.failure, duration: 2);
+      () {
+        Future<void> load() async {
+          await Future.microtask(() async {
+            await fullWeatherStateNotifier.loadFullWeather();
+          });
         }
-      }),
-      [cityStateNotifier],
+
+        load();
+
+        return null;
+      },
+      [fullWeatherStateNotifier],
     );
 
     useEffect(
-      () => fullWeatherStateNotifier.addListener((state) {
-        if (state is w.Error) {
-          showFailureSnackBar(context, failure: state.failure, duration: 2);
+      () {
+        if (fullWeatherState.fullWeather == null) {
+          return null;
         }
-      }),
-      [fullWeatherStateNotifier],
+        return cityStateNotifier.addListener((state) {
+          if (state is c.Error) {
+            showFailureSnackBar(context, failure: state.failure, duration: 2);
+          }
+        });
+      },
+      [cityStateNotifier, fullWeatherState.fullWeather],
```

Comment:
I'm not really sure about this one, but doing this:
```suggestion
      [cityStateNotifier, fullWeatherState.fullWeather == null],
```
will evaluate the function passed to `useEffect` less often (same for the below `useEffect`). However, I'm not sure if that's considered good style or not. I'll think about it a bit more and post my thoughts here.

Reply:
Okay, do it I guess. Same below.